### PR TITLE
Docs changes for v1.21.1 and v1.20.2

### PR DESCRIPTION
**Pull request by sunjayBhatia**

### CODE REVIEW

File: site/content/resources/compatibility-matrix.md [hunk 1]
```diff
@@ -135,6 +138,8 @@ __Note:__ This list of extensions was last verified to be complete with Envoy v1
 [67]: https://github.com/projectcontour/contour-operator/releases/tag/v1.20.0
 [68]: https://github.com/projectcontour/contour-operator/releases/tag/v1.20.1
 [69]: https://github.com/projectcontour/contour-operator/releases/tag/v1.21.0
+[70]: https://github.com/projectcontour/contour-operator/releases/tag/v1.21.1
+[70]: https://github.com/projectcontour/contour-operator/releases/tag/v1.20.2
```
Comment (skriss) [inline site/content/resources/compatibility-matrix.md:142]:
```suggestion
[71]: https://github.com/projectcontour/contour-operator/releases/tag/v1.20.2
```

File: site/content/resources/upgrading.md [hunk 2]
```diff
@@ -9,9 +9,76 @@ This document describes the changes needed to upgrade your Contour installation.
...
+    ```
+
+1. Once the Contour deployment has finished upgrading, update the Envoy DaemonSet:
+
+    ```bash
+    $ kubectl apply -f examples/contour/03-envoy.yaml
+    ```
+
+
+## Upgrading Contour 1.20.1 to 1.21.0
```
Comment (skriss) [inline site/content/resources/upgrading.md:81]: I guess we should update this heading too:
```suggestion
## Upgrading Contour 1.20.2 to 1.21.0
```

Comment (sunjayBhatia) [reply site/content/resources/upgrading.md]: good call!

File: site/content/resources/compatibility-matrix.md (hunk 1 above)
Comment (sunjayBhatia) [reply site/content/resources/compatibility-matrix.md]: good catch!

**Review by skriss — APPROVED** [review]

**Pull request merged**